src/values/set.ts

```diff
@@ -0,0 +1,181 @@
+import { Client } from "../client";
+import { Query, fql } from "../query-builder";
+import { QuerySuccess, QueryValue } from "../wire-protocol";
+
+/**
+ * A materialize view of a Set.
+ * @see {@link https://fqlx-beta--fauna-docs.netlify.app/fqlx/beta/reference/language/types#set}
+ */
+export class Page<T extends QueryValue> {
+  /** A materialized page of data */
+  readonly data: T[];
+  /**
+   * A pagination cursor, used to obtain additional information in the Set.
+   * If `after` is not provided, then `data` must be present and represents the
+   * last Page in the Set.
+   */
+  readonly after?: string;
+
+  constructor({ data, after }: { data: T[]; after?: string }) {
+    this.data = data;
+    this.after = after;
+  }
+}
+
+/**
+ * A un-materialize Set. Typically received when a materialized Set contains
+ * another set, the EmbeddedSet does not contain any data to avoid potential
+ * issues such as self-reference and infinite recursion
+ * @see {@link https://fqlx-beta--fauna-docs.netlify.app/fqlx/beta/reference/language/types#set}
+ */
+export class EmbeddedSet {
+  /**
+   * A pagination cursor, used to obtain additional information in the Set.
+   */
+  readonly after: string;
+
+  constructor(after: string) {
+    this.after = after;
+  }
+}
+
+/**
+ * A class to provide an iterable API for fetching multiple pages of data, given
+ * a Fauna Set
+ */
+export class SetIterator<T extends QueryValue>
+  implements AsyncGenerator<T[], void, unknown>
+{
+  readonly #generator: AsyncGenerator<T[], void, unknown>;
+
+  constructor(
+    client: Client,
+    initial: Page<T> | EmbeddedSet | (() => Promise<T>)
+  ) {
+    if (initial instanceof Function) {
+      this.#generator = generateFromThunk(client, initial);
+    } else if (initial instanceof Page || initial instanceof EmbeddedSet) {
+      this.#generator = generatePages(client, initial);
+    } else {
+      throw new TypeError(
+        "Expected 'Pageable<QueryValue> | (() => Promise<T>)', but received " +
```

Comment:
There's no such construct as `Pageable`. 

Rewrite your error message to refer to what actually exists (Page or EmbeddedSet)

If they share some interface I suppose you could have them both implement a "Pageable" interface you define as another alternative.

Reply:
Great catch, I did have a type alias for a hot second there, but this needs fixed 👍🏻